frontend/server/src/Controllers/Course.php

```diff
@@ -2318,20 +2318,17 @@ public static function getCourseSubmissionsListForSmarty(\OmegaUp\Request $r) {
             );
         }
 
-        $solvedProblems = \OmegaUp\DAO\Problems::getSolvedProblemsByUsersOfCourse(
+        $usersProblems = \OmegaUp\DAO\Problems::getProblemsByUsersInACourse(
             $r['course']
         );
         $userSolvedProblems = [];
-        foreach ($solvedProblems as $problem) {
-            $userSolvedProblems[$problem['username']][] = $problem;
-        }
-
-        $unsolvedProblems = \OmegaUp\DAO\Problems::getUnsolvedProblemsByUsersOfCourse(
-            $r['course']
-        );
         $userUnsolvedProblems = [];
-        foreach ($unsolvedProblems as $problem) {
-            $userUnsolvedProblems[$problem['username']][] = $problem;
+        foreach ($usersProblems as $userProblem) {
+            if ($userProblem['solved']) {
+                $userSolvedProblems[$userProblem['username']][] = $userProblem;
+                continue;
+            }
+            $userUnsolvedProblems[$userProblem['username']][] = $userProblem;
```

Comment:
aquí como no hay un caso de "error" obvio, se vale no usar el guard clause pattern:

```suggestion
            if ($userProblem['solved']) {
                $userSolvedProblems[$userProblem['username']][] = $userProblem;
            } else {
                $userUnsolvedProblems[$userProblem['username']][] = $userProblem;
            }
```

Reply:
Listo